contracts/examples/guards/BaseGuard.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+import {IERC165} from "../../interfaces/IERC165.sol";
+import {BaseTransactionGuard, ITransactionGuard} from "../../base/GuardManager.sol";
+import {BaseModuleGuard, IModuleGuard} from "../../base/ModuleManager.sol";
+
+/**
+ * @title BaseGuard - Inherits BaseTransactionGuard and BaseModuleGuard.
+ */
+abstract contract BaseGuard is BaseTransactionGuard, BaseModuleGuard {
+    /**
+     * @notice Checks if the contract supports an interface.
+     * @param interfaceId The interface identifier.
+     * @return True if the interfaceId matches the interfaceId of ITransactionGuard/IModuleGuard/IERC165, false otherwise.
+     */
+    function supportsInterface(bytes4 interfaceId) external view virtual override(BaseTransactionGuard, BaseModuleGuard) returns (bool) {
+        return
+            interfaceId == type(ITransactionGuard).interfaceId || // 0xe6d7a83a
+            interfaceId == type(IModuleGuard).interfaceId || // 0xe1ab3a1a
+            interfaceId == type(IERC165).interfaceId; // 0x01ffc9a7
+    }
+
+    function checkAfterExecution(bytes32, bool) external virtual override(ITransactionGuard, IModuleGuard);
```

Comment:
This function is missing NatSpec documentation. Note that it _should_ be different from the base classes (so I don't think `inheritdoc` is appropriate here) since the hash can either be a Safe transaction hash, or a module transaction has that was returned by `checkModuleTransaction`.

Actually, now that I think about it, because of the different semantics, I really do think they should be separate methods (i.e. `checkAfterExecution` and `checkAfterModuleExecution`).

Reply:
> because of the different semantics, I really do think they should be separate methods

agreed. Will update the interface

Reply:
https://github.com/safe-global/safe-smart-account/pull/758/commits/dbd83c7f8709fd9b2e39278186f43b9798d4503e